roundedbutton/src/main/java/io/intrepid/roundedbutton/RoundedButton.java

```diff
@@ -0,0 +1,128 @@
+
+            // use the default fill color if stroke isn't specified
+            int defaultFillColorRes =
+                    strokeColor == INVALID_COLOR ? R.color.rb_default_fill_color : android.R.color.transparent;
+            int defaultFillColor = ContextCompat.getColor(context, defaultFillColorRes);
+            fillColor = typedArray.getColor(R.styleable.RoundedButton_rb_fill_color, defaultFillColor);
+
+            disabledColor = typedArray.getColor(R.styleable.RoundedButton_rb_disabled_color, ContextCompat.getColor(getContext(), R.color.rb_default_disabled_color));
+
+            typedArray.recycle();
+        }
+
+        if (attrs == null || attrs.getAttributeValue(ANDROID_NAMESPACE, "gravity") == null) {
+            // default to center gravity if it's not specified
+            setGravity(Gravity.CENTER);
+        }
+
+        updateBackground();
+    }
+
+    public void setFillColor(@ColorInt int color) {
+        this.fillColor = color;
+        updateBackground();
+    }
+
+    public void setStroke(@Dimension int strokeWidth, @ColorInt int color) {
+        this.strokeWidth = strokeWidth;
+        this.strokeColor = color;
+        updateBackground();
+    }
+
+    public void setDisabledColor(@ColorInt int color) {
+        this.disabledColor = color;
+        updateBackground();
+    }
+
+    private void updateBackground() {
+        StateListDrawable stateListDrawable = new StateListDrawable();
+        Drawable defaultDrawable = createRoundedRectangleDrawable(fillColor, strokeColor, strokeWidth);
+        int darkenedFill = darkenColor(fillColor);
+        Drawable pressedDrawable = createRoundedRectangleDrawable(darkenedFill, strokeColor, strokeWidth);
+        Drawable disabledDrawable = createRoundedRectangleDrawable(disabledColor, strokeColor, strokeWidth);
+        stateListDrawable.addState(new int[] { -android.R.attr.state_enabled }, disabledDrawable);
+        stateListDrawable.addState(new int[] { android.R.attr.state_pressed }, pressedDrawable);
+        stateListDrawable.addState(StateSet.WILD_CARD, defaultDrawable);
+
+        setBackground(stateListDrawable);
+    }
+
+    private Drawable createRoundedRectangleDrawable(int fillColor, int strokeColor, int strokeWidth) {
+        GradientDrawable rectDrawable = new GradientDrawable();
+        rectDrawable.setShape(GradientDrawable.RECTANGLE);
+        if (strokeColor != INVALID_COLOR) {
+            rectDrawable.setStroke(strokeWidth, strokeColor);
+        }
+        rectDrawable.setColor(fillColor);
+        // using some really large radius here, Android will automatically clip it down to height/2
+        rectDrawable.setCornerRadius(10000);
+        return rectDrawable;
+    }
+
+    // Create a slightly darkened color for touch indicator
+    private int darkenColor(int originalColor) {
```

Comment:
darker fills would have a less visible pressed state. I just added an option to manually set the pressed state color